### PR TITLE
Patch/sgrouplayout

### PR DESCRIPTION
**Pull request by johnmay**
I think probably ready for a release after this patch. This patch primarily focusses on layout of Sgroup brackets, positional variation, and attachment points.

As noted by the IUPAC guide attachment points should be drawn as crossing wavy line.

image image

Testable example:
image

SMILES: `c1ccccc1.C*.C* |$;;;;;;;_AP1;;R5$,m:6:0.1.2.3.4.5,m:8:0.1.2.3.4.5|`
Full Code:
```java
IAtomContainer mol = new SmilesParser(SilentChemObjectBuilder.getInstance()).parseSmiles("c1ccccc1.CC.C[R5]");
        
// set attach point
AtomContainerManipulator.replaceAtomByAtom(mol, mol.getAtom(7), new PseudoAtom("_AP1"){{
    setAttachPointNum(1);
}});

// make positional variation
Sgroup pv1 = new Sgroup();
pv1.setType(SgroupType.ExtMulticenter);
pv1.addAtom(mol.getAtom(6));
pv1.addBond(mol.getBond(6));
pv1.addAtom(mol.getAtom(0));
pv1.addAtom(mol.getAtom(1));
pv1.addAtom(mol.getAtom(2));
pv1.addAtom(mol.getAtom(3));
pv1.addAtom(mol.getAtom(4));
pv1.addAtom(mol.getAtom(5));

Sgroup pv2 = new Sgroup();
pv2.setType(SgroupType.ExtMulticenter);
pv2.addAtom(mol.getAtom(8));
pv2.addBond(mol.getBond(7));
pv2.addAtom(mol.getAtom(0));
pv2.addAtom(mol.getAtom(1));
pv2.addAtom(mol.getAtom(2));
pv2.addAtom(mol.getAtom(3));
pv2.addAtom(mol.getAtom(4));
pv2.addAtom(mol.getAtom(5));

mol.setProperty(CDKConstants.CTAB_SGROUPS, Arrays.asList(pv1, pv2));
new DepictionGenerator().depict(mol).writeTo("~/tmp.svg");
```

Still more to do
  - V3000 reading improvements include Sgroups (writer supports them)
  - Multiple Group layout, this one might be tricky but should be possible

### CODE REVIEW

**Comment by johnmay** [comment]
Relevant bit from the documentation - 
image

**Comment by johnmay** [comment]
Added one more commit for handling the multiple group sgroups. Tricky to lay out as need to match substructures.

**Comment by johnmay** [comment]
Spotted something whilst writing latest blog post.

File: storage/io/src/test/java/org/openscience/cdk/io/MDLV3000ReaderTest.java [hunk 1]
```diff
@@ -117,4 +125,13 @@ public void testPseudoAtomLabels() throws Exception {
         }
     }
 
+    @Test public void positionalVariation() throws Exception {
+        MDLV3000Reader reader = new MDLV3000Reader(getClass().getResourceAsStream("multicenterBond.mol"));
+        IAtomContainer container = reader.read(new org.openscience.cdk.AtomContainer(0,0,0,0));
```
Comment (egonw) [inline storage/io/src/test/java/org/openscience/cdk/io/MDLV3000ReaderTest.java]: I always get warnings that the reader is not closed... do you have any idea if doing or not has an effect on how fast the tests run? (just wondering...)

Comment (johnmay) [inline storage/io/src/test/java/org/openscience/cdk/io/MDLV3000ReaderTest.java]: No I should have closed it there. The main problem is you can only have a certain number of file pointers open on the OS.

Comment (johnmay) [inline storage/io/src/test/java/org/openscience/cdk/io/MDLV3000ReaderTest.java]: FYI - just added a hotfix before release that resolves it.

**Pull request merged**